odpf/siren/v1/siren.proto

```diff
@@ -0,0 +1,335 @@
+syntax = "proto3";
+package odpf.siren.v1;
+
+import "google/api/annotations.proto";
+import "google/api/field_behavior.proto";
+import "google/protobuf/descriptor.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+import "validate/validate.proto";
+
+option go_package = "github.com/odpf/proton/siren";
+option java_multiple_files = true;
+option java_package = "io.odpf.proton.siren";
+option java_outer_classname = "ServiceManager";
+
+These annotations are used when generating the OpenAPI file.
+option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
+    info: {
+        version: "0.1";
+    };
+    external_docs: {
+        description: "Siren server";
+    }
+    schemes: HTTP;
+};
+
+service SirenService {
+    rpc Ping(PingRequest) returns (PingResponse) {
+        option (google.api.http) = {
+            get: "/ping"
+        };
+    }
+
+    rpc ListAlertHistory(ListAlertHistoryRequest) returns (ListAlertHistoryResponse) {
+        option (google.api.http) = {
+            get: "/history"
+        };
+    }
+
+    rpc CreateAlertHistory(CreateAlertHistoryRequest) returns (CreateAlertHistoryResponse) {
+        option (google.api.http) = {
+            post: "/history"
+            body: "*"
+        };
+    }
+
+    rpc ListWorkspaceChannels(ListWorkspaceChannelsRequest) returns (ListWorkspaceChannelsResponse) {
+        option (google.api.http) = {
+            get: "/workspaces/{workspace_name}/channels"
+        };
+    }
+    rpc ExchangeCode(ExchangeCodeRequest) returns (ExchangeCodeResponse) {
+        option (google.api.http) = {
+            post: "/oauth/slack/token",
+            body: "*"
+        };
+    }
+    rpc GetAlertCredentials(GetAlertCredentialsRequest) returns (GetAlertCredentialsResponse) {
+        option (google.api.http) = {
+            get: "/teams/{team_name}/credentials"
+        };
+    }
+
+    rpc UpdateAlertCredentials(UpdateAlertCredentialsRequest) returns (UpdateAlertCredentialsResponse) {
+        option (google.api.http) = {
+            put: "/teams/{team_name}/credentials"
+            body: "*"
+        };
+    }
+
+    rpc SendSlackNotification(SendSlackNotificationRequest) returns (SendSlackNotificationResponse) {
+        option (google.api.http) = {
+            post: "/notifications"
+            body: "*"
+        };
+    }
+
+    rpc ListRules(ListRulesRequest) returns (ListRulesResponse) {
+        option (google.api.http) = {
+            get: "/rules"
+        };
+    }
+    rpc UpdateRule(UpdateRuleRequest) returns (Rule) {
```

Comment:
Wrap this into `UpdateRuleResponse` message.

Reply:
Done